components/tabs/src/storage.rs

```diff
@@ -174,7 +174,7 @@ impl TabsStorage {
                     // Truncate the title to some limit and append ellipsis
                     // to incate that we've truncated
                     if tab.title.len() > MAX_TITLE_CHAR_LENGTH {
-                        tab.title.truncate(MAX_TITLE_CHAR_LENGTH - 3);
+                        tab.title = safe_truncate(&tab.title, MAX_TITLE_CHAR_LENGTH - 1).to_string();
```

Comment:
Is `- 1`  correct?  That depends on how many bytes that unicode character below uses.

I think this calculation should be use  `'\u{2026}'.len()` and maybe it should be inside `safe_truncate`

Reply:
you're correct on this! I updated to use the length of the ellipsis (and also moved all of that to the truncating method itself